src/Draco.Compiler/Api/Compilation.cs

```diff
@@ -204,14 +271,26 @@ public sealed class Compilation : IBinderProvider
             Diagnostics: ImmutableArray<Diagnostic>.Empty);
     }
 
+    internal ModuleSymbol GetModuleForSyntaxTree(SyntaxTree tree)
+    {
+        var filePath = Path.TrimEndingDirectorySeparator(tree.SourceText.Path?.LocalPath ?? string.Empty);
+        var rootPath = this.DeclarationTable.RootPath;
+
+        // If we don't have root path or this tree is in memory only or the tree is outside of the root, return the root module
+        if (string.IsNullOrEmpty(rootPath)
+            || string.IsNullOrEmpty(filePath)
+            || !filePath.StartsWith(rootPath)) return this.SourceModule;
+
+        var subPath = filePath[rootPath.Length..].TrimStart(Path.DirectorySeparatorChar);
+        var moduleName = Path.TrimEndingDirectorySeparator(Path.GetDirectoryName(subPath) ?? string.Empty).Replace(Path.DirectorySeparatorChar, '.');
```

Comment:
I think this code assumes that the passed in directory separator is always in the OS-default form. Shouldn't we at least normalize the root module in the Compilation ctor for this to be actually true?